manifests/server/database.pp

```diff
@@ -19,6 +20,7 @@
   Optional[String[1]] $locale           = $postgresql::server::locale,
   Boolean             $istemplate       = false,
   Hash                $connect_settings = $postgresql::server::default_connect_settings,
+  Variant[String[1], Stdlib::Port, Integer] $port = $postgresql::params::port,
```

Comment:
This means you need to always specify it when you override it in the server. Perhaps it can default to `undef` and in the body use some `pick()` logic?